[URGENT] Fix CI/CD Pipeline Failures Blocking PR #25

## 🚨 Critical Issue

The CI/CD pipeline is failing on PR #25, blocking the merge of critical security fixes for the v1.0.0 refactoring. There are two distinct failures in the GitHub Actions workflows that need immediate attention.

## Impact

- **Blocked PR**: #25 - Phase 1 security fixes (API key exposure, return type consistency)
- **Affected**: All status checks on PR #25 are failing
- **Severity**: URGENT - This blocks the entire v1.0.0 refactoring pipeline

## Error 1: Performance Test Syntax Error

### Error Message
```
Run set -euo pipefail
📊 Running baseline performance benchmarks...
file:///home/runner/work/node-grocy/node-grocy/performance-test.mjs:67
      const response = await fetch(\`\${config.baseUrl}/api/products\`);
                                   ^

SyntaxError: Invalid or unexpected token
    at compileSourceTextModule (node:internal/modules/esm/utils:340:16)
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:146:18)
    at #translate (node:internal/modules/esm/loader:431:12)
    at ModuleLoader.loadAndTranslate (node:internal/modules/esm/loader:478:27)
    at async ModuleJob._link (node:internal/modules/esm/module_job:110:19)

Node.js v20.19.1
Error: Process completed with exit code 1.
```

### Root Cause
The template literals in `performance-test.mjs` have escaped backticks (`\``) which creates invalid JavaScript syntax. This appears to be from the performance benchmarking workflow.

### Suggested Fix
Review the performance workflow file and ensure template literals use proper backticks without escape characters:
```javascript
// ❌ Invalid
const response = await fetch(\`\${config.baseUrl}/api/products\`);

// ✅ Valid
const response = await fetch(`${config.baseUrl}/api/products`);
```

## Error 2: npm Package Not Found

### Error Message
```
npm error code E404
npm error 404 Not Found - GET https://registry.npmjs.org/@typedoc%2fplugin-markdown - Not found
npm error 404
npm error 404  '@typedoc/plugin-markdown@latest' is not in this registry.
```

### Root Cause
The package name is being URL-encoded (`@typedoc%2fplugin-markdown` instead of `@typedoc/plugin-markdown`). This is happening during npm install in the CI workflow.

### Suggested Fix
1. Check if the package name is being passed through URL encoding somewhere in the workflow
2. Ensure the package name is properly escaped or quoted:
```yaml
# If in a workflow file
- run: npm install @typedoc/plugin-markdown

# Or if using quotes
- run: npm install "@typedoc/plugin-markdown"
```

## Reproduction Steps

1. Check PR #25 status checks
2. Click on failing workflow runs
3. Observe the two distinct error patterns

## Additional Context

These failures appear to be introduced by the GitOps workflows from PR #23. The workflows need to properly handle:
- JavaScript template literals without escaping backticks
- npm package names with special characters (@ and /)

## Action Items

1. [ ] Fix template literal syntax in performance-test.mjs or the workflow that generates it
2. [ ] Fix npm package name encoding in CI workflow
3. [ ] Verify all workflows handle special characters correctly
4. [ ] Re-run CI on PR #25 to confirm fixes

## Related Issues/PRs

- PR #23: GitOps implementation (introduced the workflows)
- PR #25: Phase 1 security fixes (blocked by these failures)
- Issue #26: ES module compatibility (previously fixed)
- PR #27: ES module fix (resolved similar workflow issues)

cc: @democratize-technology-code-developer 

---
*This is blocking critical security fixes for the v1.0.0 release. These workflows were working after PR #27, so this appears to be a regression or a new issue.*

--- a/.github/workflows/api-compatibility.yml
+++ b/.github/workflows/api-compatibility.yml
@@ -33,7 +33,7 @@ jobs:
         npm ci
         # Install API analysis tools
         npm list jsdoc || npm install --save-dev jsdoc@latest
-        npm list @apidevtools/json-schema-ref-parser || npm install --save-dev @apidevtools/json-schema-ref-parser@latest
+        npm list "@apidevtools/json-schema-ref-parser" || npm install --save-dev "@apidevtools/json-schema-ref-parser@latest"
 
     - name: Extract Current API Surface
       run: |

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -46,7 +46,7 @@ jobs:
         npm ci
         # Install documentation tools
         npm list typedoc || npm install --save-dev typedoc@latest
-        npm list @typedoc/plugin-markdown || npm install --save-dev @typedoc/plugin-markdown@latest
+        npm list "@typedoc/plugin-markdown" || npm install --save-dev "@typedoc/plugin-markdown@latest"
         npm list jsdoc || npm install --save-dev jsdoc@latest
 
     - name: Setup Documentation Environment

--- a/.github/workflows/performance.yml
+++ b/.github/workflows/performance.yml
@@ -196,7 +196,7 @@ jobs:
             
             try {
               // Simple HTTP request test
-              const response = await fetch(\`\${config.baseUrl}/api/products\`);
+              const response = await fetch(`${config.baseUrl}/api/products`);
               await response.json();
               
               const duration = performance.now() - start;
@@ -229,7 +229,7 @@ jobs:
           
           try {
             const promises = Array.from({ length: concurrency }, async (_, i) => {
-              const response = await fetch(\`\${config.baseUrl}/api/stock\`);
+              const response = await fetch(`${config.baseUrl}/api/stock`);
               return response.json();
             });
             
@@ -265,7 +265,7 @@ jobs:
           // Perform multiple operations to test memory handling
           for (let i = 0; i < 20; i++) {
             try {
-              const response = await fetch(\`\${config.baseUrl}/api/products\`);
+              const response = await fetch(`${config.baseUrl}/api/products`);
               await response.json();
             } catch (e) {
               // Continue testing even if some requests fail
@@ -299,12 +299,12 @@ jobs:
             console.log('Performance test results:', JSON.stringify(results, null, 2));
             require('fs').writeFileSync('performance-results.json', JSON.stringify(results, null, 2));
             
-            console.log(\`\nSummary:
-            - Average latency: \${results.summary.avgLatency.toFixed(2)}ms
-            - Max latency: \${results.summary.maxLatency.toFixed(2)}ms
-            - Min latency: \${results.summary.minLatency.toFixed(2)}ms
-            - Total requests: \${results.summary.totalRequests}
-            - Errors: \${results.summary.errorsCount}\`);
+            console.log(`\nSummary:
+            - Average latency: ${results.summary.avgLatency.toFixed(2)}ms
+            - Max latency: ${results.summary.maxLatency.toFixed(2)}ms
+            - Min latency: ${results.summary.minLatency.toFixed(2)}ms
+            - Total requests: ${results.summary.totalRequests}
+            - Errors: ${results.summary.errorsCount}`);
           })
           .catch(error => {
             console.error('Performance test failed:', error);
@@ -349,7 +349,7 @@ jobs:
               metric: 'avgLatency',
               current: current.avgLatency,
               threshold: thresholds.avgLatency,
-              message: \`Average latency \${current.avgLatency.toFixed(2)}ms exceeds threshold \${thresholds.avgLatency}ms\`
+              message: `Average latency ${current.avgLatency.toFixed(2)}ms exceeds threshold ${thresholds.avgLatency}ms`
             });
           }
           
@@ -358,7 +358,7 @@ jobs:
               metric: 'maxLatency',
               current: current.maxLatency,
               threshold: thresholds.maxLatency,
-              message: \`Max latency \${current.maxLatency.toFixed(2)}ms exceeds threshold \${thresholds.maxLatency}ms\`
+              message: `Max latency ${current.maxLatency.toFixed(2)}ms exceeds threshold ${thresholds.maxLatency}ms`
             });
           }
           
@@ -367,7 +367,7 @@ jobs:
               metric: 'errorRate',
               current: errorRate,
               threshold: thresholds.errorRate,
-              message: \`Error rate \${(errorRate * 100).toFixed(2)}% exceeds threshold \${(thresholds.errorRate * 100)}%\`
+              message: `Error rate ${(errorRate * 100).toFixed(2)}% exceeds threshold ${(thresholds.errorRate * 100)}%`
             });
           }
           
@@ -390,14 +390,14 @@ jobs:
         const comparison = comparePerformance();
         
         console.log('Performance Comparison Results:');
-        console.log(\`Status: \${comparison.status}\`);
-        console.log(\`Average Latency: \${comparison.summary.avgLatency.toFixed(2)}ms\`);
-        console.log(\`Max Latency: \${comparison.summary.maxLatency.toFixed(2)}ms\`);
-        console.log(\`Error Rate: \${comparison.summary.errorRate.toFixed(2)}%\`);
+        console.log(`Status: ${comparison.status}`);
+        console.log(`Average Latency: ${comparison.summary.avgLatency.toFixed(2)}ms`);
+        console.log(`Max Latency: ${comparison.summary.maxLatency.toFixed(2)}ms`);
+        console.log(`Error Rate: ${comparison.summary.errorRate.toFixed(2)}%`);
         
         if (comparison.violations.length > 0) {
           console.log('\nPerformance Violations:');
-          comparison.violations.forEach(v => console.log(\`  - \${v.message}\`));
+          comparison.violations.forEach(v => console.log(`  - ${v.message}`));
         }
         
         if (comparison.hasRegression) {

